src/Workspaces/Core/Portable/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.cs

```diff
@@ -95,11 +96,21 @@ protected IEnumerable<ITypeSymbol> ExpandParamsParameter(IParameterSymbol parame
 
         public IEnumerable<ITypeSymbol> InferTypes(SemanticModel semanticModel, int position, CancellationToken cancellationToken)
         {
-            return CreateTypeInferrer(semanticModel, cancellationToken).InferTypes(position);
+            return CreateTypeInferrer(semanticModel, cancellationToken).InferTypes(position).Select(t => t.InferredType);
         }
 
         public IEnumerable<ITypeSymbol> InferTypes(SemanticModel semanticModel, SyntaxNode expression, CancellationToken cancellationToken)
         {
+            return CreateTypeInferrer(semanticModel, cancellationToken).InferTypes(expression as TExpressionSyntax).Select(t => t.InferredType);
```

Comment:
Worth pre-allocating a collection and copying into it to avoid multiple evaluations of the `Select`?